File: python/01_ocr_extract.py
"""Extract text from a PDF or image using DeepRead OCR."""

import os
import sys
import time
import requests

API_KEY = os.environ.get("DEEPREAD_API_KEY")
BASE = "https://api.deepread.tech"

if not API_KEY:
    print("Set DEEPREAD_API_KEY in your environment or .env file")
    sys.exit(1)

file_path = sys.argv[1] if len(sys.argv) > 1 else None
if not file_path:
    print("Usage: python 01_ocr_extract.py <file.pdf>")
    sys.exit(1)

headers = {"X-API-Key": API_KEY}

# Submit document
with open(file_path, "rb") as f:
    resp = requests.post(f"{BASE}/v1/process", headers=headers, files={"file": f})
    resp.raise_for_status()

job = resp.json()
job_id = job["id"]
print(f"Submitted: {job_id} (status: {job['status']})")

# Poll with backoff
delay = 5
while True:
    time.sleep(delay)
    result = requests.get(f"{BASE}/v1/jobs/{job_id}", headers=headers).json()
    status = result["status"]
    print(f"  Status: {status}")

    if status == "completed":
        res = result["result"]
        print("\n--- Extracted Text ---")
        print(res.get("text_preview") or res.get("text", "")[:2000])
        if result.get("preview_url"):
            print(f"\nShareable preview: {result['preview_url']}")
        break
    elif status == "failed":
        print(f"Failed: {result.get('error', 'Unknown error')}")
        break

    delay = min(delay * 1.5, 15)

File: node/01_ocr_extract.mjs
// Extract text from a PDF or image using DeepRead OCR

import { readFileSync } from "fs";

const API_KEY = process.env.DEEPREAD_API_KEY;
const BASE = "https://api.deepread.tech";
const filePath = process.argv[2];

if (!API_KEY) { console.error("Set DEEPREAD_API_KEY"); process.exit(1); }
if (!filePath) { console.error("Usage: node 01_ocr_extract.mjs <file.pdf>"); process.exit(1); }

const form = new FormData();
form.append("file", new Blob([readFileSync(filePath)]), filePath.split("/").pop());

const { id: jobId } = await fetch(`${BASE}/v1/process`, {
  method: "POST",
  headers: { "X-API-Key": API_KEY },
  body: form,
}).then(r => r.json());

console.log(`Submitted: ${jobId}`);

let delay = 3000;
while (true) {
  await new Promise(r => setTimeout(r, delay));
  const result = await fetch(`${BASE}/v1/jobs/${jobId}`, {
    headers: { "X-API-Key": API_KEY },
  }).then(r => r.json());

  console.log(`  Status: ${result.status}`);

  if (result.status === "completed") {
    console.log("\n--- Extracted Text ---");
    console.log((result.text_preview || result.text || "").slice(0, 2000));
    if (result.preview_url) console.log(`\nFull preview: ${result.preview_url}`);
    break;
  } else if (result.status === "failed") {
    console.error(`Failed: ${result.error || "Unknown error"}`);
    break;
  }
  delay = Math.min(delay * 1.5, 15000);
}
